java-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/resolution/typesolvers/JavaParserTypeSolver.java

```diff
@@ -129,26 +142,24 @@ private CompilationUnit parse(File srcFile) throws FileNotFoundException {
             }
 
             File srcFile = new File(filePath);
-            try {
-                CompilationUnit compilationUnit = parse(srcFile);
-                Optional<com.github.javaparser.ast.body.TypeDeclaration<?>> astTypeDeclaration = Navigator.findType(compilationUnit, typeName);
-                if (astTypeDeclaration.isPresent()) {
-                    return SymbolReference.solved(JavaParserFacade.get(this).getTypeDeclaration(astTypeDeclaration.get()));
+            {
```

Comment:
I like this technique. It is underused and should be in every developer's repetoire. It should ring the "maybe we need to extract some methods" bell though.